In [1]:
import pandas as pd

In [5]:
data=pd.read_csv("downloads/train.csv")

In [7]:
data.head(10)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   
5            6         0       3   
6            7         0       1   
7            8         0       3   
8            9         1       3   
9           10         1       2   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   
5                                   Moran, Mr. James    male   NaN      0   
6                            McCarthy, Mr. Timothy J    male  54.0      0   
7                     Palsson, Master. Gosta Leonard    male   2.0      3   
8  Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S  
5      0            330877   8.4583   NaN        Q  
6      0             17463  51.8625   E46        S  
7      1            349909  21.0750   NaN        S  
8      2            347742  11.1333   NaN        S  
9      0            237736  30.0708   NaN        C

## Cleaning the data


In [60]:
data.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age              0
SibSp            0
Parch            0
Ticket           0
Fare             0
Embarked         2
familysize       0
fareperperson    0
dtype: int64

<p>We can check that age, cabin and Embarked has null values hence we need to fix them</p>
<OL>
    <li>replace null age with median(age) of passengers.
</li>
    <li>
drop cabin column as it has too many null values
</li>
    <li>
we will keep embarked unchanged as it has no effect on results.</li>
</OL>

In [9]:
data["Age"]=data["Age"].fillna(data["Age"].median())

In [11]:
data= data.drop(columns=["Cabin"])

In [13]:
gender_survival=data.groupby("Sex")["Survived"].mean()

In [55]:
Survived_gender=data.groupby("Sex")["Survived"].sum()

In [56]:
total_gender=data.groupby("Sex")["Survived"].count()

In [58]:
print(Survived_gender)

Sex
female    233
male      109
Name: Survived, dtype: int64


In [59]:
print(total_gender)

Sex
female    314
male      577
Name: Survived, dtype: int64


## Analysis keypoints
<UL>
    <li>
        Female survival rate was higher(74.2%) compared to males(18.9%).
    </li>
    <li>
        233 females out of 314 total females survived, while only 109 men survived out of 577 onboard.
    </li>
</UL>

In [35]:
data["familysize"]=data["SibSp"]+data["Parch"]+1

In [62]:
data.groupby("familysize")["Fare"].mean().reset_index()

familysize       Fare
0           1  21.242689
1           2  49.894129
2           3  39.692482
3           4  54.864510
4           5  58.094453
5           6  73.722727
6           7  29.366667
7           8  46.900000
8          11  69.550000

## Analysis keypoints
<UL>
    <li>
        Families of 6 people paid highest average price \$73.72.
    </li>
    <li>
        Solo travelers paid lowest price per family i.e \$21.24.
    </li>
    <li>
        Trend shows that Families with high numbers of members paid higher price per family however price per person is lower. 
    </li>
</UL>